### PR TITLE
fix: dtoss 5069 valid primary care provider rules now changed

### PR DESCRIPTION
**Pull request by SamRobinson75684**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by andrewmorris43 — APPROVED** [review]

**Review by Warren-Pitterson — APPROVED** [review]

File: application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs [hunk 1]
```diff
@@ -26,7 +32,7 @@ public DbLookupValidationBreastScreening(IDbConnection IdbConnection)
     /// </summary>
     /// <param name="primaryCareProvider">The participant's primary care provider.</param>
     /// <returns>bool, whether or not the GP practice code exists in the DB.<returns>
-    public bool ValidatePrimaryCareProvider(string primaryCareProvider)
+    public bool PrimaryCareProviderExists(string primaryCareProvider)
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs:35]: Why was this renamed? I know it reads better in the rules but method names should be based around a verb

File: application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs [hunk 2]
```diff
@@ -106,28 +113,50 @@ public bool ValidateLanguageCode(string languageCode) {
...
                 {
                     while (reader.Read())
                     {
-                        return reader.GetInt32(0) == 1;
+                        postingCategory = reader["POSTING_CATEGORY"].ToString();
+                        isCurrentPostingInDB = reader.GetInt32(1) == 1;
                     }
+                }
+
+                var CurrentPostingDoesNotExistInDB = currentPosting != null && !isCurrentPostingInDB && !validatePostingCategories(currentPosting);
```
Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs:141]: I would prefer if this logic was part of the rule, i.e create a method for validating the posting category and then do this as part of a local param and include it in the rule expression.

Comment (will-larkin-nhs) [inline application/CohortManager/src/Functions/Shared/Data/Database/DbLookupValidationBreastScreening.cs:141]: Also should be using string.IsNullOrEmpty here

**Review by Warren-Pitterson — APPROVED** [review]

**Review by will-larkin-nhs — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -58,3 +58,4 @@ application/CohortManager/SampleData/
 recreate_local_settings.sh
 
 .history
+application/CohortManager/Set-up/SampleData/CaaSSchema02.json
```
Comment (andrewmorris43) [inline .gitignore:61]: Why is this file now ignored in here?
